txtdirectmain/run.go

```diff
@@ -1,179 +1,166 @@
-		return caddy.CaddyfileFromPipe(os.Stdin, serverType)
-	}
-
-	contents, err := ioutil.ReadFile(conf)
-	if err != nil {
-		return nil, err
-	}
-
-	return caddy.CaddyfileInput{
-		Contents:       contents,
-		Filepath:       conf,
-		ServerTypeName: serverType,
-	}, nil
-}
-
-// defaultLoader loads the TXTDirect config from the current working directory.
-func defaultLoader(serverType string) (caddy.Input, error) {
-	contents, err := ioutil.ReadFile(caddy.DefaultConfigFile)
-	if err != nil {
-		if os.IsNotExist(err) {
-			return nil, nil
-		}
-		return nil, err
-	}
-	return caddy.CaddyfileInput{
-		Contents:       contents,
-		Filepath:       caddy.DefaultConfigFile,
-		ServerTypeName: serverType,
-	}, nil
-}
-
-// showVersion prints the version that is starting.
-func showVersion() {
-	fmt.Print(versionString())
-	fmt.Print(releaseString())
-	if devBuild && gitShortStat != "" {
-		fmt.Printf("%s\n%s\n", gitShortStat, gitFilesModified)
-	}
-}
-
-// versionString returns the TXTDirect version as a string.
-func versionString() string {
-	return fmt.Sprintf("%s-%s\n", caddy.AppName, caddy.AppVersion)
-}
-
-// releaseString returns the release information related to TXTDirect version:
-// <OS>/<ARCH>, <go version>, <commit>
-// e.g.,
-// linux/amd64, go1.8.3, a6d2d7b5
-func releaseString() string {
-	return fmt.Sprintf("%s/%s, %s, %s\n", runtime.GOOS, runtime.GOARCH, runtime.Version(), GitCommit)
-}
-
-// setVersion figures out the version information
-// based on variables set by -ldflags.
-func setVersion() {
-	// A development build is one that's not at a tag or has uncommitted changes
-	devBuild = gitTag == "" || gitShortStat != ""
-
-	// Only set the appVersion if -ldflags was used
-	if gitNearestTag != "" || gitTag != "" {
-		if devBuild && gitNearestTag != "" {
-			appVersion = fmt.Sprintf("%s (+%s %s)", strings.TrimPrefix(gitNearestTag, "v"), GitCommit, buildDate)
-		} else if gitTag != "" {
-			appVersion = strings.TrimPrefix(gitTag, "v")
-		}
-	}
-}
-
-// Flags that control program flow or startup
-var (
-	conf    string
-	version bool
-)
-
-// Build information obtained with the help of -ldflags
-var (
-	appVersion = "(untracked dev build)" // inferred at startup
-	devBuild   = true                    // inferred at startup
-
-	buildDate        string // date -u
-	gitTag           string // git describe --exact-match HEAD 2> /dev/null
-	gitNearestTag    string // git describe --abbrev=0 --tags HEAD
-	gitShortStat     string // git diff-index --shortstat
-	gitFilesModified string // git diff-index --name-only HEAD
-
-	// Gitcommit contains the commit where we built TXTDirect from.
-	GitCommit string
-)
-
-func getFlags() error {
-	versionFlag := flag.Lookup("version")
-	if versionFlag != nil {
-		versionVal, err := strconv.ParseBool(versionFlag.Value.String())
-		if err != nil {
-			return fmt.Errorf("Coudln't parse the -version flag: %s", err.Error())
-		}
-		version = versionVal
-	}
-
-	return nil
-}
+// func init() {
```

Comment:
That workaround doesn't work anymore I assume. Probably needs to be reimplemented. Any reason to comment instead of removing the code? We do have versioning.

Reply:
Leftover, should've been deleted but missed it.